src/partition/repart.c

```diff
@@ -113,6 +118,8 @@ static PagerFlags arg_pager_flags = 0;
 static bool arg_legend = true;
 static void *arg_key = NULL;
 static size_t arg_key_size = 0;
+static void *arg_certificate = NULL;
+static size_t arg_certificate_size = 0;
```

Comment:
Why not make this `static X509 *arg_certificate = NULL;`?

Reply:
Because then we have to add HAVE_OPENSSL ifdeffery everywhere. With this approach, the ifdeffery is nicely contained to 2 functions.

Reply:
i think this could be contained. i.e. in openssl-util.h add:

```c
#ifndef HAVE_OPENSSL
typedef struct X509 X509;
#endif
```

That way we can just use the types also when openssl is not available, and don't have to bother.

(and maybe we need some NOP implementation of X509_free() too, that asserts the input pointer is NULL and does nothing else. i.e. if ew have no openssl, then we couldn#t possibly have an X509 object, and thus all pointers must be NULL)